src/js/control-bar/progress-control/mouse-time-display.js

```diff
@@ -21,6 +22,19 @@ class MouseTimeDisplay extends Component {
   constructor(player, options) {
     super(player, options);
 
+    if (options.playerOptions &&
+        options.playerOptions.controlBar &&
+        options.playerOptions.controlBar.progressControl &&
+        options.playerOptions.controlBar.progressControl.keepTooltipsInside) {
+      this.keepTooltipsInside = options.playerOptions.controlBar.progressControl.keepTooltipsInside;
+    }
+
+    if (this.keepTooltipsInside) {
+      this.el().innerHTML = `<div class='vjs-time-tooltip'>`;
+      this.tooltip = this.el().querySelector('.vjs-time-tooltip');
```

Comment:
I wonder if this would be preferable to adding to the DOM then querying it?
```js
this.tooltip = Dom.createEl('div', {className: 'vjs-time-tooltip'});
this.el().appendChild(this.tooltip);
```